cli/types/cypress.d.ts

```diff
@@ -3048,6 +3048,14 @@ declare namespace Cypress {
      * @default 'top'
      */
     scrollBehavior: scrollBehaviorOptions
+    /**
+     * Indicates whether Cypress should strip CSP header directives from the application under test.
+     * NOTE: When this option is set to `false`, Cypress **will** continue to strip the `frame-ancestors` directive
+     * because it prevents loading the target document into an iframe. Please see the documentation for more information.
+     * @see https://on.cypress.io/configuration#stripCspDirectives
+     * @default true
+     */
+    stripCspDirectives: boolean | string | string[],
```

Comment:
```suggestion
    /**
     * Indicates whether Cypress should strip CSP header directives from the application under test.
     * NOTE: Setting this value to 'none' or not stripping `frame-ancestors`
     * may prevent Cypress from functioning because its presence prevents 
     * loading the target document into an iframe. Please see the documentation 
     * for more information.
     * @see https://on.cypress.io/configuration#stripCspDirectives
     * @default 'all'
     */
    stripCspDirectives: 'all' | 'none' | string | string[],
```

A couple changes: 

- Use only strings with some enumerated values instead of mixing booleans and strings. `true` is now "all", `false` is now "none"
- It seems inconsistent to strip `frame-ancestors` when false/none is specified, but not when specific directives are specified. I'd rather give users control of this and have the caveat be consistent amongst non-"all" values.

Reply:
@chrisbreiding So what you are saying is that 'none' should allow all CSP headers...even the ones that prevent Cypress from running? If that's the case, then why 'none' and not an empty string?

As an adendum to that question, why allow a static attribute like 'none' to implement behavior that blocks Cypress from running on sites with 'frame-ancestors'? Is it because it would error people's tests immediately and force them to specify 'frame-ancestors'?

Reply:
> Imagine that there is a new CSP directive that is released that inhibits Cypress functionality. That would require all users to now add a new directive to their stripCspDirectives array. If there were an enum that automatically stripped the problematic directives, Cypress could add the directive to the unsupportedCSPDirectives, and all users who used the enum value would get that header stripped.

I hadn't thought of that. I agree we should do away with "none" and have an enum value that equates to what `false` does. What about "minimum" since it would be stripping the minimum number of directives for Cypress to function?

Reply:
@chrisbreiding Sounds good. I'll update using "minimum"